### ISSUE TITLE
Enhancement request: version of signer for old-school algos (i.e RSA + NIST curves)

### ISSUE THREAD

**Issue by udf2457** [tillitis/tkey-device-signer#6]
Thanks for your work on tkey.

It would be nice to see a version of signer that supports old-school algos such as RSA and the NIST p-curves.

I know Ed2559 is awesome and all that, but unfortunately there are applications that don't yet support Ed25519.  For example CNCF project SPIFFE/SPIRE [only supports EC or RSA key types](https://github.com/spiffe/spire-plugin-sdk/blob/a2e5ba68e76045c0ca332264af935ea8c3a59f99/proto/spire/plugin/agent/keymanager/v1/keymanager.proto#L123).  Using tkey in conjunction with SPIFFE/SPIRE could be a very interesting use-case for tkey.

Sadly whilst I am a proficient Go programmer, my C knowledge is non existent.  And so sadly because tkey apps need to be written in C I would be too worried about messing up C crypto myself !

**Comment by secworks** [comment]
Good idea! We need to look at suitable libraries to use and exact type of signing to support.

**Comment by jku** [comment]
There are no doubt hardware limitations that still prevent some algorithms but #28 would make the protocol usable with larger keys and signatures
